server/app/middlewares/websocket_backend.rb

```diff
@@ -121,7 +121,10 @@ def find_node_by_node_token(node_id, node_token, init_attrs)
     if !node_by_id
       # atomically initialize the node_id
       created_node = HostNode.where(:id => node.id, :node_id => nil)
-        .find_one_and_update(:$set => {node_id: node_id, **init_attrs})
+        .find_one_and_update(
+          :$set => {node_id: node_id, **init_attrs},
```

Comment:
`init_attrs` seems a bit funny because it's currently just one attribute?

Reply:
It's also used for `find_node_by_grid_token`, and that handles both of the attrs via just `$set`.